src/main/java/org/jabref/gui/maintable/OpenExternalFileAction.java

```diff
@@ -41,38 +54,51 @@ public OpenExternalFileAction(DialogService dialogService, StateManager stateMan
-            final List<BibEntry> selectedEntries = stateManager.getSelectedEntries();
+            if (entry == null) {
+                final List<BibEntry> selectedEntries = stateManager.getSelectedEntries();
 
-            List<LinkedFileViewModel> linkedFileViewModelList = new LinkedList<>();
-            LinkedFileViewModel linkedFileViewModel;
+                List<LinkedFileViewModel> linkedFileViewModelList = new LinkedList<>();
+                LinkedFileViewModel linkedFileViewModel;
 
-            for (BibEntry entry:selectedEntries) {
-                for (LinkedFile linkedFile:entry.getFiles()) {
-                    linkedFileViewModel = new LinkedFileViewModel(
-                            linkedFile,
-                            entry,
-                            databaseContext,
-                            Globals.TASK_EXECUTOR,
-                            dialogService,
-                            preferencesService.getXmpPreferences(),
-                            preferencesService.getFilePreferences(),
-                            ExternalFileTypes.getInstance());
+                for (BibEntry entry : selectedEntries) {
+                    for (LinkedFile linkedFile : entry.getFiles()) {
+                        linkedFileViewModel = new LinkedFileViewModel(
+                                linkedFile,
+                                entry,
+                                databaseContext,
+                                Globals.TASK_EXECUTOR,
+                                dialogService,
+                                preferencesService.getXmpPreferences(),
+                                preferencesService.getFilePreferences(),
+                                ExternalFileTypes.getInstance());
 
-                    linkedFileViewModelList.add(linkedFileViewModel);
+                        linkedFileViewModelList.add(linkedFileViewModel);
+                    }
                 }
-            }
 
-            // ask the user when detecting # of files > FILES_LIMIT
-            if (linkedFileViewModelList.size() > FILES_LIMIT) {
-                boolean continueOpening = dialogService.showConfirmationDialogAndWait(Localization.lang("Opening large number of files"),
-                        Localization.lang("You are about to open %0 files. Continue?", Integer.toString(linkedFileViewModelList.size())),
-                        Localization.lang("Continue"), Localization.lang("Cancel"));
-                if (!continueOpening) {
-                    return;
+                // ask the user when detecting # of files > FILES_LIMIT
+                if (linkedFileViewModelList.size() > FILES_LIMIT) {
+                    boolean continueOpening = dialogService.showConfirmationDialogAndWait(Localization.lang("Opening large number of files"),
+                            Localization.lang("You are about to open %0 files. Continue?", Integer.toString(linkedFileViewModelList.size())),
```

Comment:
I think I added an overload a while ago where integer.toString should no be longer necessary as param

Reply:
Indeed